packages/capacitor-plugin/android/src/main/kotlin/com/outsystems/capacitor/plugins/geolocation/GeolocationPlugin.kt

```diff
@@ -0,0 +1,338 @@
+package com.outsystems.capacitor.plugins.geolocation
+
+import android.Manifest
+import android.os.Build
+import androidx.activity.result.contract.ActivityResultContracts
+import com.getcapacitor.JSObject
+import com.getcapacitor.PermissionState
+import com.getcapacitor.Plugin
+import com.getcapacitor.PluginCall
+import com.getcapacitor.PluginMethod
+import com.getcapacitor.annotation.CapacitorPlugin
+import com.getcapacitor.annotation.Permission
+import com.getcapacitor.annotation.PermissionCallback
+import com.google.android.gms.location.LocationServices
+import com.outsystems.plugins.osgeolocation.controller.OSGLOCController
+import com.outsystems.plugins.osgeolocation.model.OSGLOCException
+import com.outsystems.plugins.osgeolocation.model.OSGLOCLocationOptions
+import com.outsystems.plugins.osgeolocation.model.OSGLOCLocationResult
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+
+@CapacitorPlugin(
+    name = "GeolocationPlugin",
+    permissions = [Permission(
+        strings = [Manifest.permission.ACCESS_COARSE_LOCATION, Manifest.permission.ACCESS_FINE_LOCATION],
+        alias = GeolocationPlugin.LOCATION_ALIAS
+    ), Permission(
+        strings = [Manifest.permission.ACCESS_COARSE_LOCATION],
+        alias = GeolocationPlugin.COARSE_LOCATION_ALIAS
+    )]
+)
+class GeolocationPlugin : Plugin() {
+
+    private lateinit var controller: OSGLOCController
+    private lateinit var coroutineScope: CoroutineScope
+    private val watchingCalls: MutableMap<String, PluginCall> = mutableMapOf()
+
+    companion object {
+        const val LOCATION_ALIAS: String = "location"
+        const val COARSE_LOCATION_ALIAS: String = "coarseLocation"
+    }
+
+    override fun load() {
+        super.load()
+
+        coroutineScope = CoroutineScope(Dispatchers.Main)
+        val activityLauncher = activity.registerForActivityResult(
+            ActivityResultContracts.StartIntentSenderForResult()
+        ) { result ->
+            CoroutineScope(Dispatchers.Main).launch {
+                controller.onResolvableExceptionResult(result.resultCode)
+            }
+        }
+
+        this.controller = OSGLOCController(
+            LocationServices.getFusedLocationProviderClient(context),
+            activityLauncher
+        )
+
+    }
+
+    override fun handleOnDestroy() {
+        super.handleOnDestroy()
```

Comment:
Maybe also clear `watchingCalls` to prevent memory leaks?

Reply:
I believe that isn't necessary, as the Android system will automatically take care of clearing the map when the app is closed